Source/Relativity.Test.Helpers/Relativity.Test.Helpers/Mail/MailTrapMailHelper.cs

```diff
@@ -0,0 +1,141 @@
+using System.Collections.Generic;
+using System.Net.Http;
+
+namespace Relativity.Test.Helpers.Mail
+{
+	public class MailTrapMailHelper : IMailHelper
+	{
+		public Dictionary<string, string> RequestHeaders { get; set; }
+		public string BaseApiUrl => "https://mailtrap.io/";
+
+		public MailTrapMailHelper(string apiKey)
+		{
+			RequestHeaders = new Dictionary<string, string> { { "Api-Token", apiKey } };
+		}
+
+		/// <summary>
+		/// Gets all the inboxes on your account
+		/// </summary>
+		/// <returns></returns>
+		public List<IMailInboxModel> GetInboxes()
+		{
+			List<IMailInboxModel> inboxes = new List<IMailInboxModel>();
+
+			using (HttpClient client = new HttpClient())
+			{
+				client.BaseAddress = new Uri(BaseApiUrl);
+
+				foreach (var header in RequestHeaders)
+				{
+					client.DefaultRequestHeaders.Add(header.Key, header.Value);
+				}
+
+				string apiEndpoint = "api/v1/inboxes/";
+
+				HttpResponseMessage response = client.GetAsync(apiEndpoint).Result;
+				string json = response.Content.ReadAsStringAsync().Result;
+
+				inboxes.AddRange(JsonConvert.DeserializeObject<List<MailTrapInboxModel>>(json));
+			}
+
+			return inboxes;
+		}
+
+		/// <summary>
+		/// Gets all the messages from a particular inbox in your account
+		/// </summary>
+		/// <param name="inbox">Used to grab the Id to use in the URL for the API call</param>
+		/// <returns></returns>
+		public List<IMailMessageModel> GetMessagesInInbox(IMailInboxModel inbox)
+		{
+			List<IMailMessageModel> messages = new List<IMailMessageModel>();
+
+			using (HttpClient client = new HttpClient())
+			{
+				client.BaseAddress = new Uri(BaseApiUrl);
+
+				foreach (var header in RequestHeaders)
+				{
+					client.DefaultRequestHeaders.Add(header.Key, header.Value);
+				}
+
+				string apiEndpoint = $"api/v1/inboxes/{inbox.Id}/messages";
+
+				HttpResponseMessage response = client.GetAsync(apiEndpoint).Result;
+				string json = response.Content.ReadAsStringAsync().Result;
+
+				messages.AddRange(JsonConvert.DeserializeObject<List<MailTrapMessageModel>>(json));
+			}
+
+			return messages;
+		}
+
+		/// <summary>
+		/// Returns a specific message as HTML (in string format)
+		/// </summary>
+		/// <param name="inbox">Used to grab the Id to use in the URL for the API call</param>
+		/// <param name="messageId">MailTrapMessageModel.id is the source of this</param>
+		/// <returns></returns>
+		public IMailMessageModel GetMessage(IMailInboxModel inbox, int messageId)
+		{
+			IMailMessageModel message;
+
+			using (HttpClient client = new HttpClient())
+			{
+				client.BaseAddress = new Uri(BaseApiUrl);
+
+				foreach (var header in RequestHeaders)
+				{
+					client.DefaultRequestHeaders.Add(header.Key, header.Value);
+				}
+
+				string apiEndpoint = $"api/v1/inboxes/{inbox.Id}/messages/{messageId}/body.html";
+
+				HttpResponseMessage response = client.GetAsync(apiEndpoint).Result;
+
+				string messageResult = response.Content.ReadAsStringAsync().Result;

```

Comment:
can we check if HTTP status is success before reading the content?

Reply:
Updated in latest commit